syntax = "proto3";
import "google/protobuf/empty.proto";

package lidar;

message Point3 {
    float x=1;
    float y=2;
    float z=3;
    optional float r=4;
    optional float g=5;
    optional float b=6;
}

message PointCloud3 { 
    repeated Point3 points = 1;
    optional string entity_name = 2;
    optional uint64 timestamp = 3;
}

message IMUData {
    float ax = 1;
    float ay = 2;
    float az = 3;
    float gx = 4;
    float gy = 5;
    float gz = 6;
    optional uint64 timestamp = 7;
}

// TODO RENAME!
service LidarService {
    rpc getScan(google.protobuf.Empty) returns (stream PointCloud3);
    rpc getImu(google.protobuf.Empty) returns (stream IMUData);
}